jscomp/depends/depends_post_process.ml

```diff
@@ -49,23 +49,32 @@ let read_deps fn =
 
   return_arr 
 
+type compilation_kind_t = Js | Bytecode | Native
 
 (* TODO: Don't touch the .d file if nothing changed *)
-let handle_bin_depfile oprefix  (fn : string) index : unit = 
+let handle_bin_depfile 
+  oprefix
+  ~compilation_kind
+  (fn : string)
+  index : unit = 
+  let suffix_cmixo = match compilation_kind with
+  | Js       -> Literals.suffix_cmj
+  | Bytecode -> Literals.suffix_cmo
```

Comment:
note bytecode module b depends on a, you only need record `b.cmo : a.cmi` but record `b.cmo : a.cmo` is not that bad (slightly slower)